app/controllers/sessions_controller.rb

```diff
@@ -45,34 +52,21 @@ class SessionsController < ApplicationController
   end
 
   post '/sign_up' do
-    username = params[:username]
+    password_confirmation = params[:password_confirmation]
     password = params[:password]
-    first_name = params[:first_name]
-    second_name = params[:second_name]
 
-    @user = User.new(
-      username: username,
-      password: password,
-      first_name: first_name,
-      second_name: second_name,
-      password_confirmation: params[:password_confirmation]
-    )
+    @error_messages = ["Password confirmation doesn't match password"] unless password_confirmation == password
 
-    if @user.valid?
-      @user.save
-      session[:user_id] = @user.id
-      redirect '/homepage'
-    else
-      @error_messages = @user.errors.full_messages
-      erb :'user/sign_up_page'
-    end
+    create_user('/homepage', { username: params[:username],
+                               password: params[:password],
+                               first_name: params[:first_name],
+                               second_name: params[:second_name],
+                               password_confirmation: params[:password_confirmation],
+                               password_status: 'permanent' }, :'user/sign_up_page')
```

Comment:
That to be more readable 
could be in this way
opts = {username: params[],......}
create_user('page', opts, :'page')